curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/6448e475f904ad580cd7e722.md

```diff
@@ -0,0 +1,170 @@
+---
+id: 6448e475f904ad580cd7e722
+title: Step 4
+challengeType: 0
+dashedName: step-4
+---
+
+# --description--
+
+Now that your `checkUserInput()` function is set up for testing, you can use an event listener to call the function when users click the `Convert` button.
+
+Chain the `.addEventListener()` method to `convertBtn`. The event listener should listen for `click` events and take a reference to the `checkUserInput` function as a callback. Remember that function references are not called with parentheses.
+
+Once that's done, whenever you click the `Convert` button, the value of the number input should be logged to the console.
+
+# --hints--
+
+You should call the `.addEventListener()` method on your `convertBtn` element.
+
+```js
+assert.match(code, /convertBtn\s*\.\s*addEventListener\s*\(/);
+```
+
+Your `.addEventListener()` method should listen for the `click` event.
+
+```js
+assert.match(code, /convertBtn\s*\.\s*addEventListener\s*\(\s*('|"|`)click\1\s*,/);
```

Comment:
the comma should be optional, because campers technically followed the directinos with this answer here`convertBtn.addEventListener('click')`

```suggestion
assert.match(code, /convertBtn\s*\.\s*addEventListener\s*\(\s*('|"|`)click\1\s*,?/);
```